### DISCUSSION TITLE
Intermittent Image Conversion Slowness - Illustrator files

### DISCUSSION THREAD

**Discussion by trythisone35** [ImageMagick/ImageMagick#3345] [Help]
We have noticed that certain Adobe Illustrator files take exponentially longer to process than others.   Although the files are big (40-60MB), not all behave this way.   The majority of them take less than 30 seconds.   However, some can take 10 minutes or longer.   The processing time is repeatable on these "problem files".   Is there anything in an Illustrator file that can contribute to such long processing times (i.e. embedded tiffs, layers, etc)?

Thank you in advance.

**Comment by fmw42** [comment]
What was your command?  What is your IM version and platform?  What is your version of Ghostscript? 

Density contributes directly to the output rasterized file size and thus the time to process.

**Reply by trythisone35** [reply]
Did some testing on the 54MB original file and found the one feature in the artwork that made the conversion time increase by 4x (4 min vs 1 min).  Interestingly, there's a similar feature that doesn't cause any trouble.   Is it possible to determine what might be problematic about this feature?   

We do have our density set high (600).   However, we've been running that way for years with no trouble.  



[problemfeature.zip](https://github.com/ImageMagick/ImageMagick/files/6090768/problemfeature.zip)

**Comment by snibgo** [comment]
As fmw42 suggests, the problem may be in the image size, as measured by the number of pixels. If this is too large for your available memory, IM will store the pixel cache on disk, which is much slower than memory.

**Comment by snibgo** [comment]
Please answer fmw42's questions:
> What was your command? What is your IM version and platform?

This command, on IM v7.0.8-64 and Ghostscript 9.19, works fine:
```
magick -verbose -density 10 problemfeature.ai x.png
```
The output is 149*269 pixels. So at density 600, we would expect 144 million pixels. When you run the command with "-density 600", is there high usage of memory or disk?

**Reply by trythisone35** [reply]
Hi there, i am using the following command:

magick -density 600 problemfeature.ai  x.png
When doing density of 100 it completes in about 2.5 minutes.
I'm using IM v7.0.11 and Ghostscript 9.53.3.

I actually am unable to get a density of 600 to finish on an .ai file even without the problem feature.   You might be on to something regarding the disk usage.   It seems to try to write and then gives up.  Waits and tries again.  Not sure if that makes sense.   Are the limitations for memory/disk on the program side or hardware side?   In other words, if I had enough memory/space would the conversion still face a limitation when the output appears to big.

Btw, thank you for your help.  I'm not much of an expert so i appreciate you trying to point me in the right direction.

**Reply by trythisone35** [reply]
Had a friend run the same file on his box (RAM 32GB @ 3200mhz 1TB ssd, 4.04 GHz 8 core compared to my 4GB RAM) and he was able to get an output file after about 11 minutes.  So, not surprisingly I suppose, that's a huge factor.   We are doing some more testing.

**Comment by snibgo** [comment]
144 million pixels at 4 channels, 2 bytes/channel/pixel, needs 1.152 GB. If you have only 4 GB then that is likely to be a problem. How much _free_ memory do you have? Are you also running other programs, such as web browsers? If so, then try it without no other programs running.

**Reply by trythisone35** [reply]
I will try that.   The only puzzling item is that we had been running along just fine for several years with an older version of IM and GS.   Would anything in the newer versions impact performance?

**Reply by dlemstra** [reply]
It is possible that you are now using an HDRI build?
